kafka-rest/src/main/java/io/confluent/kafkarest/GroupMetadataObserver.java

```diff
@@ -0,0 +1,276 @@
+                      .all().get(initTimeOut, TimeUnit.MILLISECONDS).get(groupId).members();
+      if (summariesOpt.size() > 0) {
+        final Set<TopicName> result = new HashSet<>();
+        for (MemberDescription eachSummary : summariesOpt) {
+          for (TopicPartition topicPartition : eachSummary.assignment().topicPartitions()) {
+            result.add(new TopicName(topicPartition.topic()));
+          }
+        }
+        log.debug("Get topic list {}", result);
+        final boolean needPartOfData = offsetOpt.nonEmpty()
+                && countOpt.nonEmpty()
+                && countOpt.get() > 0;
+        if (!needPartOfData) {
+          return result;
+        } else {
+          final scala.collection.Set<TopicName> slice =
+                  JavaConverters.asScalaSetConverter(result).asScala()
+                          .slice(offsetOpt.get(), offsetOpt.get() + countOpt.get()).toSet();
+          return JavaConverters.setAsJavaSetConverter(slice).asJava();
+        }
+      } else {
+        return Collections.emptySet();
+      }
+    } else {
+      return Collections.emptySet();
+    }
+  }
+
+  /**
+   * <p>Get consumer group description</p>
+   *
+   * @param groupId - group name
+   * @return description of consumer group
+   *     (all consumed topics with all partition offset information)
+   */
+  public ConsumerEntity getConsumerGroupInformation(String groupId) throws Exception {
+    return getConsumerGroupInformation(groupId, Option.<String>empty(),
+            Option.<Integer>empty(), Option.<Integer>empty());
+  }
+
+  /**
+   * <p>Get consumer group description</p>
+   *
+   * @param groupId   - group name
+   * @param topic     - topic name for filter - default empty topic name
+   * @param offsetOpt - offset for TopicPartitionEntity
+   *                  collection for each consumer member for paging
+   * @param countOpt  - count of elements TopicPartitionEntity
+   *                  collection for each consumer member for paging
+   * @return description of consumer group
+   */
+  public ConsumerEntity getConsumerGroupInformation(String groupId, Option<String> topic,
+                                                    Option<Integer> offsetOpt,
+                                                    Option<Integer> countOpt) throws Exception {
+    if (Option.apply(adminClient).nonEmpty()) {
+      final ConsumerGroupDescription consumerGroupSummary =
+              adminClient.describeConsumerGroups(Collections.singleton(groupId))
+                      .all().get(initTimeOut, TimeUnit.MILLISECONDS).get(groupId);
+      final Collection<MemberDescription> summaries =
+              consumerGroupSummary.members();
+      if (summaries.size() > 0) {
+        log.debug("Get summary list {}", summaries);
+        try (KafkaConsumer kafkaConsumer = createConsumer(groupId, config)) {
+          final List<TopicPartitionEntity> confluentTopicPartitions = new ArrayList<>();
```

Comment:
Could we rename this variable? I don't see the relation to `confluent`. Did you mean `consumerTopicPartitions`?